addons/mail/models/mail_thread.py

```diff
@@ -888,6 +888,10 @@ def message_route(self, message, message_dict, model=None, thread_id=None, custo
         if not isinstance(message, EmailMessage):
             raise TypeError('message must be an email.message.EmailMessage at this point')
         catchall_alias = self.env['ir.config_parameter'].sudo().get_param("mail.catchall.alias")
+        catchall_domain = self.env["ir.config_parameter"].sudo().get_param("mail.catchall.domain")
+        catchall_domains_allowed = self.env["ir.config_parameter"].sudo().get_param("mail.catchall.domain.allowed")
+        if catchall_domain and catchall_domains_allowed:
+            catchall_domains_allowed = catchall_domains_allowed.split(',') + [catchall_domain]
```

Comment:
We could also lower catchall_domain, to lower everything

Reply:
Indeed, but I notice a weird behavior: The field `value` of a system parameter is `required`, but no value is set by default for `mail.catchall.domain`. As soon as we change it, we cannot clean it except by removing the record. I guess it might be ok since we should normally check first if the param exists (with `get_param`)...

However, if someone modify it and put anything in it (even just a space), it will not be cleaned/sanitized.
Question: Should we do a validation of the `mail.catchall.domain` (like for `mail.catchall.domain.allowed`) or just do `.strip().lower()` (and then check if there is a value) when using it?

Reply:
indeed we generally don't check if it contains space, ... we just check we have a value. Would say that it can be left as it is, as anyway you copy-paste your email domain in it. A comma-separated list (like this new param) is more subject to issues as people could make mistakes (double comma, spaces, ...).

TL;DR: would just do .strip().lower() here as suggested and let it untouched anywhere else